admin-dev/themes/new-theme/js/components/form/form-object-mapper.ts

```diff
@@ -369,6 +377,30 @@ export default class FormObjectMapper {
     }
   }
 
+  /**
+   * We need a custom checking method for equality, we don't use strict equality on purpose because it would result
+   * into a potential infinite loop if type doesn't match, which can easily happen with a number value in a text input.
+   *
+   * And we also try to see if both values have the same Number value, this avoids forcing a number input value when
+   * it's not written exactly the same way (like pending zeros)
```

Comment:
These are implementation details that belong in the method's body, not in the header.

The header documentation is meant to describe what the method does (i.e. what mutation it performs if it's a command, and what information it returns if it's a query) so that you don't need to reverse engineer it, hence the importance of describing the parameters and return values beyond the types.